dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -217,6 +272,219 @@ public static String digest(final String text) {
+     * This method takes a Map of Optional<char[]> As this is a middle ground object representation
+     * that can be mapped from a saved  AppSecrets or an incoming SecretForm
+     * if the param isn't included in the map it means it wasn't sent..
+     * if the param was sent empty that would be represented as an empty optional.
+     * I'm using optional since null vales on map triggers warnings
+     * @param params
+     * @param appDescriptor
+     */
+    public static void validateForSave(final Map<String, Optional<char[]>> params,
+            final AppDescriptor appDescriptor) {
+
+        //Param/Property names are case sensitive.
+        final Map<String, ParamDescriptor> appDescriptorParams = appDescriptor.getParams();
+
+        for (final Entry<String, ParamDescriptor> descriptorParam : appDescriptorParams.entrySet()) {
+            final String describedParamName = descriptorParam.getKey();
+            //initialize to null so it is not found in the params map it means it wasn't sent.
+            char[] input = null;
+            if (params.containsKey(describedParamName)) {
+                // if the key is found then verify if there's an actual value or else null
+                final Optional<char[]> optionalChars = params.get(describedParamName);
+                input = optionalChars.orElse(null);
+            }
+            if (descriptorParam.getValue().isRequired() && (input == null || isNotSet(input))) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Param `%s` is marked required in the descriptor but does not come with a value.",
+                                describedParamName
+                        )
+                );
+            }
+
+            if (null == input) {
+                //Param wasn't sent but it doesn't matter since it isn't required.
+                Logger.debug(AppsAPIImpl.class, () -> String
+                        .format("Non required param `%s` was set.",
+                                describedParamName));
+                continue;
+            }
+
+            if (Type.BOOL.equals(descriptorParam.getValue().getType()) && UtilMethods
+                    .isSet(input)) {
+                final String asString = new String(input);
+                final boolean bool = (asString.equalsIgnoreCase(Boolean.TRUE.toString())
+                        || asString.equalsIgnoreCase(Boolean.FALSE.toString()));
+                if (!bool) {
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Can not convert value `%s` to type BOOL for param `%s`.",
+                                    asString, describedParamName
+                            )
+                    );
+                }
+            }
+
+            if (Type.SELECT.equals(descriptorParam.getValue().getType()) && UtilMethods
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'UtilMethods.isSet' due to existing static import 'com.dotmarketing.util.UtilMethods.isSet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)